### DISCUSSION TITLE
Why is the golang version so accurate to the minor version?

### DISCUSSION THREAD

**Discussion by liuchangyan** [etcd-io/etcd#16571] [Q&A] [answered]
I am trying to compile etcd manually with this command `DRY_RUN=false ./scripts/release.sh 3.5.9`
and my golang version is `go1.19.13`, and it has reported this error: 
`Current go version is go1.19.13, but etcd v3.5.9 requires go1.19.9 (see .go-version).`

I’ve seen this file https://github.com/etcd-io/etcd/blob/v3.5.9/.go-version and I want to know why is the golang version so accurate to the minor version?

And I found it is ok to compile (scripts/build.sh) the master code with the latest golang version and it has no error, I want to know if it is ok for me to use the latest golang version to compile etcd and is there some unknown problem that I'll met?

**Comment by jmhbnz** [comment, marked as answer]
Hey @liuchangyan - Thanks for your question.

Yes, the etcd project does specify an exact golang patch release in `.go-version`.  We discussed this in https://github.com/etcd-io/etcd/issues/15674 and the project consensus was explicitly defining the patch version was desired.

If you are trying to recompile an existing release it would probably be best to align to that same golang patch version, depending on your goals for the recompile.  However I wouldn't expect any issues doing so with a higher patch release version.

Just a quick note, for the upcoming release `3.5.10` the project has moved to golang 1.20, refer https://github.com/etcd-io/etcd/issues/16393.
